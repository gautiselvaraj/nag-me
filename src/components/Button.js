import React from 'react';
import styled, {css} from 'styled-components';

const ButtonStyles = css`
  box-sizing: border-box;
  font-size: .85rem;
  line-height: ${props => props.small ? '1.5rem' : '2rem'};
  text-decoration: none;
  text-transform: uppercase;
  transition-duration: .5s;
  transition-property: background-color;
`;

const CircleButtonStyles = css`
  border-radius: 50%;
  height: ${props => props.small ? '1.5rem' : '2rem'};
  text-align: center;
  width: ${props => props.small ? '1.5rem' : '2rem'};

  i {
    position: relative;
    top: 2px;
  }
`;

const Button = styled.button`
  background-color: ${props => props.reset ? 'transparent' : props.theme.main};
  border: ${props => props.reset ? '0' : `1px solid ${props => props.theme.mainDarken}`};
  color: ${props => props.reset ? props.theme.main : props.theme.white};
  cursor: pointer;
  display: ${props => props.block ? 'block' : 'inline-block'};
  padding: ${props => props.reset || props.circle ? '0' : '0 .75rem'};
  ${props => props.block ? 'width: 100%' : ''};
  ${props => props.reset ? '' : ButtonStyles};
  ${props => props.circle ? CircleButtonStyles : ''};

  &:hover {
    background-color: ${props => props.reset ? 'transparent' : props.theme.mainLighten};
    color: ${props => props.reset ? props.theme.mainLighten : props.theme.white};
  }
`;

const Link = Button.withComponent('a');

export default ({children, link, ...otherProps}) => (
  link ?
  <Link {...otherProps}>{children}</Link> :
  <Button {...otherProps}>{children}</Button>
);
