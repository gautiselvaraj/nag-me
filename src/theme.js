export default {
  main:           '#214154',
  mainDarken:     '#083148',
  mainLighten:    '#215a7b',
  black:          '#111111',
  white:          '#f6f6f6',
  greyLightest:   '#efefef',
  greyLighter:    '#dddddd',
  greyLight:      '#a5a5a5',
  grey:           '#898989',
  greyDark:       '#777777',
  greyDarker:     '#595959',
  greyDarkest:    '#222',
}
