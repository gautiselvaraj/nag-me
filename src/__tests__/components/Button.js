import React from 'react';
import { mount } from 'enzyme';
import Button from '../../components/Button';
import theme from '../../theme';
import 'jest-styled-components';

describe('<Button />', () => {
  it('should match snapshot and have correct text, element & background-color', () => {
    const button = mount(<Button theme={theme}>Button</Button>);
    expect(button).toMatchSnapshot();
    expect(button.text()).toBe('Button');
    expect(button.find('button').length).toBe(1);
    expect(button).toHaveStyleRule('background-color', theme.main);
  });

  it('should match snapshot and have correct element & background-color based on reset prop', () => {
    const button = mount(
      <Button reset theme={theme}>
        Button
      </Button>
    );
    expect(button).toMatchSnapshot();
    expect(button.find('button').length).toBe(1);
    expect(button).toHaveStyleRule('background-color', 'transparent');
  });

  it('should match snapshot and have correct element & border-radius based on circle prop', () => {
    const button = mount(
      <Button circle theme={theme}>
        Button
      </Button>
    );
    expect(button).toMatchSnapshot();
    expect(button.find('button').length).toBe(1);
    expect(button).toHaveStyleRule('border-radius', '50%');
  });

  it('should match snapshot and have correct element based on link prop', () => {
    const button = mount(
      <Button link href="https://www.gauti.info/nag-me" theme={theme}>
        Link
      </Button>
    );
    expect(button).toMatchSnapshot();
    expect(button.find('a').length).toBe(1);
  });
});
