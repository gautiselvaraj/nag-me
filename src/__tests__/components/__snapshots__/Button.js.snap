// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> circle button should match snapshot 1`] = `
.c0 {
  background-color: #214154;
  border: 1px solid #08283B;
  color: #f6f6f6;
  cursor: pointer;
  display: inline-block;
  padding: 0;
  box-sizing: border-box;
  font-size: 0.85rem;
  line-height: 2rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition-duration: 0.5s;
  transition-duration: 0.5s;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  border-radius: 50%;
  height: 2rem;
  text-align: center;
  width: 2rem;
}

.c0 i {
  position: relative;
  top: 2px;
}

.c0:hover {
  background-color: #3B5B6E;
  color: #f6f6f6;
}

<button
  className="c0"
>
  Button
</button>
`;

exports[`<Button /> default button should match snapshot 1`] = `
.c0 {
  background-color: #214154;
  border: 1px solid #08283B;
  color: #f6f6f6;
  cursor: pointer;
  display: inline-block;
  padding: 0 .75rem;
  box-sizing: border-box;
  font-size: 0.85rem;
  line-height: 2rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition-duration: 0.5s;
  transition-duration: 0.5s;
  -webkit-transition-property: background-color;
  transition-property: background-color;
}

.c0:hover {
  background-color: #3B5B6E;
  color: #f6f6f6;
}

<button
  className="c0"
>
  Button
</button>
`;

exports[`<Button /> link button should match snapshot 1`] = `
.c0 {
  background-color: #214154;
  border: 1px solid #08283B;
  color: #f6f6f6;
  cursor: pointer;
  display: inline-block;
  padding: 0 .75rem;
  box-sizing: border-box;
  font-size: 0.85rem;
  line-height: 2rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition-duration: 0.5s;
  transition-duration: 0.5s;
  -webkit-transition-property: background-color;
  transition-property: background-color;
}

.c0:hover {
  background-color: #3B5B6E;
  color: #f6f6f6;
}

<a
  className="c0"
  href="https://www.gauti.info/nag-me"
>
  Link
</a>
`;

exports[`<Button /> reset button should match snapshot 1`] = `
.c0 {
  background-color: transparent;
  border: 0;
  color: #214154;
  cursor: pointer;
  display: inline-block;
  padding: 0;
}

.c0:hover {
  background-color: transparent;
  color: #3B5B6E;
}

<button
  className="c0"
>
  Button
</button>
`;
